lib/java/src/org/apache/thrift/Option.java

```diff
@@ -54,6 +56,19 @@ public T or(T other) {
             return other;
         }
     }
+
+    /**
+     * Turn this Option into Java 8 Optional type
+     * @return Java 8+ Optional Type
+     */
+    public Optional<T> toOptional() {
+        if (isDefined()) {
+            return Optional.of(get());
+        } else {
+            return Optional.empty();
+        }
```

Comment:
Could also do:

```suggestion
        return Optional.ofNullable(isDefined() ? get() : null);
```

Reply:
I'd keep this as is because it is more explicit while the proposed change results in two if-comparison, one in here and another in the optional class itself